presto-spi/src/main/java/com/facebook/presto/spi/type/IpAddressType.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.type;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import io.airlift.slice.Slice;
+
+import static com.facebook.presto.spi.type.TypeSignature.parseTypeSignature;
+
+public class IpAddressType
+        extends AbstractFixedWidthType
+{
+    public static final IpAddressType IPADDRESS = new IpAddressType();
+    public static final int IP_ADDRESS_SIZE = 16;
+
+    private IpAddressType()
+    {
+        super(parseTypeSignature(StandardTypes.IPADDRESS), Slice.class, IP_ADDRESS_SIZE);
+    }
+
+    @Override
+    public boolean isComparable()
+    {
+        return true;
+    }
+
+    @Override
+    public boolean isOrderable()
+    {
+        return true;
+    }
+
+    @Override
+    public boolean equalTo(Block leftBlock, int leftPosition, Block rightBlock, int rightPosition)
+    {
+        return compareTo(leftBlock, leftPosition, rightBlock, rightPosition) == 0;
+    }
+
+    @Override
+    public int compareTo(Block leftBlock, int leftPosition, Block rightBlock, int rightPosition)
+    {
+        return leftBlock.compareTo(leftPosition, 0, getFixedSize(), rightBlock, rightPosition, 0, getFixedSize());
+    }
+
+    @Override
+    public long hash(Block block, int position)
+    {
+        return block.hash(position, 0, getFixedSize());
+    }
+
+    @Override
+    public Object getObjectValue(ConnectorSession session, Block block, int position)
+    {
+        if (block.isNull(position)) {
+            return null;
+        }
+        return new SqlIpAddress(block.getSlice(position, 0, getFixedSize()).getBytes());
```

Comment:
Since we want to format this properly when returning this to the client, use `String` for the object representation. `InetAddress` will not work as it doesn't use RFC 5952.